packages/codegen/src/commands.ts

```diff
@@ -0,0 +1,338 @@
+// Copyright 2022 Cognite AS
+import path from 'path';
+import { promises as fs } from 'fs';
+import { OpenApiDocument } from './openapi';
+import { VersionFileManager } from './versionfile';
+import {
+  FromPathOption,
+  PackageOption,
+  ServiceOption,
+  versionFileDirectoryPath,
+  VersionOption,
+} from './utils';
+import { CodeGen, PassThroughFilter, ServiceNameFilter } from './codegen';
+import {
+  ConfigScopeOption,
+  ConfigurationManager,
+  ConfigurationOptions,
+} from './configuration';
+
+type VersionFileOptions = PackageOption & Partial<ServiceOption>;
+type ConfigFileOptions = PackageOption & ServiceOption;
+type ConfigFileUpdateOptions = PackageOption &
+  ServiceOption &
+  VersionOption &
+  Partial<ConfigScopeOption>;
+
+export class VersionFileCommand {
+  public create = async (
+    options: VersionFileOptions &
+      Partial<VersionOption> &
+      Partial<FromPathOption>
+  ) => {
+    const directory = await versionFileDirectoryPath(options);
+    const vfm = new VersionFileManager({ directory: directory, ...options });
+    const versionfileExists = await vfm.exists();
+    if (versionfileExists) {
+      throw new Error(
+        `Versionfile already exists - did nothing (did you mean "update"?)`
+      );
+    }
+
+    await this.updateFile(vfm, options);
+  };
+
+  public update = async (
+    options: VersionFileOptions &
+      Partial<VersionOption> &
+      Partial<FromPathOption>
+  ) => {
+    const directory = await versionFileDirectoryPath(options);
+    const vfm = new VersionFileManager({ directory: directory, ...options });
+    const versionfileExists = await vfm.exists();
+    if (!versionfileExists) {
+      throw new Error(
+        `Versionfile does not exist - did nothing (did you mean "create"?)`
+      );
+    }
+
+    await vfm.delete();
+    await this.updateFile(vfm, options);
+  };
+
+  public delete = async (options: VersionFileOptions) => {
+    const directory = await versionFileDirectoryPath(options);
+    const vfm = new VersionFileManager({ directory: directory });
+    const versionfileExists = await vfm.exists();
+    if (!versionfileExists) {
+      throw new Error(`Versionfile does not exist`);
+    }
+
+    await vfm.delete();
+  };
+
+  private updateFile = async (
+    vfm: VersionFileManager,
+    options: Partial<FromPathOption>
+  ): Promise<void> => {
+    const download = async (): Promise<OpenApiDocument> => {
+      if (typeof options['from-path'] !== 'undefined') {
+        return await vfm.downloadFromPath({ path: options['from-path'] });
+      }
+      return await vfm.downloadFromURL();
+    };
+
+    const document = await download();
+    await vfm.write(document);
```

Comment:
nit: If you can I'd recommend using ternary operator over control flow - it often simplifies things (and notice how to check for undefined):

```suggestion
    const document =
      options['from-path'] !== undefined
        ? await vfm.downloadFromPath({ path: options['from-path'] })
        : await vfm.downloadFromURL();

    await vfm.write(document);
```